object/scw.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2018-present Juicedata Inc.
+
+package object
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+type scw struct {
+	s3client
+}
+
+func (s *scw) String() string {
+	return fmt.Sprintf("scw://%s", s.s3client.bucket)
+}
+
+func (s *scw) Create() error {
+	if _, err := s.List("", "", 1); err == nil {
+		return nil
+	}
+	_, err := s.s3.CreateBucket(&s3.CreateBucketInput{Bucket: &s.bucket})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case s3.ErrCodeBucketAlreadyExists:
+				err = nil
+			case s3.ErrCodeBucketAlreadyOwnedByYou:
+				err = nil
+			}
+		}
+	}
+	return err
+}
+
+func newScw(endpoint, accessKey, secretKey string) (ObjectStorage, error) {
+	uri, err := url.ParseRequestURI(endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid endpoint %s: %s", endpoint, err)
+	}
+	ssl := strings.ToLower(uri.Scheme) == "https"
+	hostParts := strings.Split(uri.Host, ".")
+	bucket := hostParts[0]
+	region := hostParts[2]
+	endpoint = uri.Host[len(bucket)+1:]
+
+	if accessKey == "" {
+		accessKey = os.Getenv("SCW_ACCESS_KEY")
+	}
+	if secretKey == "" {
+		secretKey = os.Getenv("SCW_SECRET_KEY")
+	}
```

Comment:
I added this, like in the `OBS` provider. I also added the corresponding line in the readme.